internal/pkg/service/stream/storage/statistics/repository/stats_put.go

```diff
@@ -0,0 +1,61 @@
+package repository
+
+import (
+	"context"
+	"sync"
+
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/level"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/storage/statistics"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+// Put creates or updates slices statistics records in the level.Local.
+func (r *Repository) Put(ctx context.Context, stats []statistics.PerSlice) (err error) {
+	ctx, span := r.telemetry.Tracer().Start(ctx, "keboola.go.buffer.storage.statistics.Repository.Put")
+	defer span.End(&err)
+
+	var currentTxn *op.TxnOp[op.NoResult]
+	var allTxn []*op.TxnOp[op.NoResult]
+	addTxn := func() {
+		currentTxn = op.Txn(r.client)
+		allTxn = append(allTxn, currentTxn)
+	}
+
+	// Merge multiple put operations into one transaction
+	i := 0
+	for _, v := range stats {
+		if i == 0 || i >= putMaxStatsPerTxn {
+			i = 0
```

Comment:
This could be optimized using `i` as part of for loop and `%` operator if I am right. The code was just moved but I would fix that.

Reply:
https://github.com/keboola/keboola-as-code/pull/1824